### PR TITLE
feat(console,phrases): add SSO connectors list page

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. Add SSO connectors list page.
2. Reorg some SSO connector type definitions to `@logto/schemas` so that the console can also rely on those types, which keeps the SSOT convention.
3. Update SSO connectors libraries methods and APIs to support pagination for `GET /api/sso-connectors` API.

## Testing

Tested locally, will add UT once the console flow is ready.
image
image

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/core/src/sso/types/index.ts [hunk 1]
```diff
@@ -11,12 +11,3 @@ export abstract class SingleSignOn {
...
 }
-
-export enum SsoProviderName {
-  OIDC = 'OIDC',
-  SAML = 'SAML',
-}
-
-export type SupportedSsoConnector = Omit<SsoConnector, 'providerName'> & {
-  providerName: SsoProviderName;
-};
```
Comment (simeng-li) [inline packages/core/src/sso/types/index.ts:22]: Better keep the definition of SsoProviderName within the SSO factories' directory.

Comment (darcyYe) [reply packages/core/src/sso/types/index.ts]: Updated.

File: packages/core/src/libraries/sso-connector.ts [hunk 2]
```diff
@@ -11,16 +11,22 @@ export type SsoConnectorLibrary = ReturnType<typeof createSsoConnectorLibrary>;
...
+      count,
+      entities.filter((connector): connector is SupportedSsoConnector =>
+        isSupportedSsoConnector(connector)
+      ),
+    ];
   };
 
-  const getAvailableSsoConnectors = async () => {
-    const connectors = await getSsoConnectors();
+  const getAvailableSsoConnectors = async (limit?: number, offset?: number) => {
```
Comment (simeng-li) [inline packages/core/src/libraries/sso-connector.ts:28]: should return all.

File: packages/core/src/routes/sso-connector/index.ts [hunk 3]
```diff
@@ -117,12 +117,17 @@ export default function singleSignOnRoutes<T extends AuthedRouter>(...args: Rout
...
       response: ssoConnectorWithProviderConfigGuard.array(),
       status: [200],
     }),
     async (ctx, next) => {
-      const connectors = await getSsoConnectors();
+      const { limit, offset, disabled: paginationDisabled } = ctx.pagination;
+      const [totalCount, connectors] = paginationDisabled
+        ? await getSsoConnectors()
+        : await getSsoConnectors(limit, offset);
+      ctx.pagination.totalCount = totalCount;
```
Comment (simeng-li) [inline packages/core/src/routes/sso-connector/index.ts:130]: Feels like we could just always enable the pagination, keep the logic simple.

**Review by simeng-li — APPROVED** [review]
core package LGTM, please wait for AC approvals.

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/sso-connector/index.ts (hunk 3 above)
Comment (darcyYe) [reply packages/core/src/routes/sso-connector/index.ts:130]: Per offline discussion, in the current stage, our customers may not have too many SSO connectors but adding the pagination logic is simple here.

**Pull request merged**